concordium-sdk/src/main/java/com/concordium/sdk/cis2/events/Cis2EventWithMetadata.java

```diff
@@ -0,0 +1,64 @@
+package com.concordium.sdk.cis2.events;
+
+import com.concordium.sdk.cis2.Cis2Error;
+import com.concordium.sdk.requests.BlockQuery;
+import com.concordium.sdk.transactions.Hash;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.Objects;
+
+/**
+ * An event emitted from a CIS2 compliant contract with some metadata.
+ */
+@EqualsAndHashCode
+@ToString
+@Getter
+public class Cis2EventWithMetadata {
+
+    /**
+     * The event of a successfully executed transaction.
+     * This is only present if the transaction was successfully executed.
+     */
+    private final Cis2Event event;
+
+    /**
+     * The error of a failed transaction.
+     * This is only present if the tranasction failed.
+     */
+    private final Cis2Error error;
+
+    /**
+     * A flag that indicates whether the event retains an
+     * {@link Cis2Event} (as a result of a successfully executed transaction)
+     * or a {@link Cis2Error} as a result of a failed transaction.
+     */
+    private final boolean isSuccessfull;
```

Comment:
Can't you check that by comparison with `null`?

Reply:
Yes. That's also how the field is set. 

I can remove the field and simply have a function if that is what you suggest.